pkg/controller/util.go

```diff
@@ -1087,16 +1087,22 @@ func getDefaultAccessModes(c client.Client, storageClass *storagev1.StorageClass
 	return nil, errors.Errorf("no accessMode defined on StorageProfile for %s StorageClass", storageClass.Name)
 }
 
-func volumeSize(c client.Client, storage *cdiv1.StorageSpec, volumeMode *v1.PersistentVolumeMode) (*resource.Quantity, error) {
+func volumeSize(c client.Client, spec cdiv1.DataVolumeSpec, volumeMode *v1.PersistentVolumeMode) (*resource.Quantity, error) {
 	// resources.requests[storage] - just copy it to pvc,
-	requestedSize, found := storage.Resources.Requests[v1.ResourceStorage]
+	requestedSize, found := spec.Storage.Resources.Requests[v1.ResourceStorage]
+
 	if !found {
-		return nil, errors.Errorf("Datavolume Spec is not valid - missing storage size")
+		// Storage size can be empty when using PVC source
+		if spec.Source.PVC != nil {
```

Comment:
what you check here is basicly `isClone`. I think it would be more readable to state:
```
// Storage size can be empty when cloning
```
or even better to do 
```
isClone := spec.Source.PVC != nil
if isClone ...
```

but I think that is just the matter of style.

Reply:
I agree about that, though maybe is a little bit redundant to create a variable for just that condition?

Reply:
the advocates of clean code might say that this variable is explaining things better than any comment so is not redundant